compute_tools/src/extension_server.rs

```diff
@@ -180,7 +180,19 @@ pub async fn download_extension(
 // Create extension control files from spec
 pub fn create_control_files(remote_extensions: &RemoteExtSpec, pgbin: &str) {
     let local_sharedir = Path::new(&get_pg_config("--sharedir", pgbin)).join("extension");
-    for ext_data in remote_extensions.extension_data.values() {
+    for (ext_name, ext_data) in remote_extensions.extension_data.iter() {
+        // Check if extension is present in public or custom.
+        // If not, then it is not allowed to be used by this compute.
+        if let Some(public_extensions) = &remote_extensions.public_extensions {
```

Comment:
NIT:

Can it be sth like this in order to decrease the number of nested conditions? 

```rust
remote_extensions.public_extensions.as_ref().or(remote_extensions.custom_extensions.as_ref()).map_or(false, |extension| extension.contains(ext_name))
```

Reply:
If I read this line correctly, this will only check `public_extensions`, if its present.
But we need to check both arrays.